### PR TITLE
TEZEX UI

### PR DESCRIPTION
**Pull request by SiddharthV1**
Tezex UI -

Add Liquidity
Remove liquidity,
Swap

along with slippage support for the relevant components

https://user-images.githubusercontent.com/424818/230388612-63c3a543-a445-4a13-a30c-428e488761b1.mov

### CODE REVIEW

File: V2/tezex/src/App.tsx [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
+			<Layout>
+				<Routes>
+					<Route path="/Home" element={<Home />} />
+					<Route path="/Analytics" element={<Analytics />} />
+					<Route path="/About" element={<About />} />
+				</Routes>
+			</Layout>
+	</header>
+</Router>
+*/
```
Comment (gdsoumya) [inline V2/tezex/src/App.tsx]: maybe we should clean up these comments if they won't be used

File: V2/tezex/src/components/addLiquidity/index.tsx_ [hunk 2]
```diff
@@ -0,0 +1,187 @@
+import { FC, useState, useEffect } from "react";
```
Comment (gdsoumya) [inline V2/tezex/src/components/addLiquidity/index.tsx_]: is this file needed I see `.tsx_` as the extension?

File: V2/tezex/src/components/swap/style.tsx [hunk 3]
```diff
@@ -0,0 +1,325 @@
+const classes = {
```
Comment (gdsoumya) [inline V2/tezex/src/components/swap/style.tsx]: I see we have 2 style files one with js and tsx extension are both needed? We should consolidate all styles into a single files

File: V2/tezex/src/contexts/network.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	apiKey: "ab682065-864a-4f11-bc77-0ef4e9493fa1",
+	addresses: {
+		sirs: {
+			address: "KT1AafHA1C1vk959wvHWBispY9Y2f3fxBUUo",
+		},
+		tzbtc: {
+			address: "KT1PWx2mnDueood7fEmfbBDKx1D9BAnnXitn",
+			dex: { sirius: "KT1TxqZ8QtKvLu3V3JH7Gx58n7Co8pgtpQU5" },
+		},
+	},
```
Comment (gdsoumya) [inline V2/tezex/src/contexts/network.tsx]: we should read this from either .env file or a json config file

File: V2/tezex/src/pages/About/index.tsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { FC} from "react";
```
Comment (gdsoumya) [inline V2/tezex/src/pages/About/index.tsx]: what are these empty files for? don't think these are rendering anything

**Review by gdsoumya — DISMISSED** [review]
Saw quiet a few files with commented code we should clean that up

File: V2/tezex/src/pages/removeLiquidity/index.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+	removeLiquidity,
+} from "../../functions/liquidityBaking";
+import { Transact } from "../../components/ui/elements/Buttons";
+
+import { TokenKind } from "../../types/general";
+
+export interface IRemoveLiquidity {
+	children: null;
+}
+export const RemoveLiquidity: FC = (props) => {
```
Comment (anonymoussprocket) [inline V2/tezex/src/pages/removeLiquidity/index.tsx]: the entire function is commented out, is it needed?

File: V2/tezex/src/pages/Home/index.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { AddLiquidity } from "../../components/addLiquidity";
+import { RemoveLiquidity } from "../../components/removeLiquidity";
+
+import Grid2 from "@mui/material/Unstable_Grid2"; // Grid version 2
+type HomePaths = "swap" | "add" | "remove";
+
+export interface IHome {
+	path: HomePaths;
+}
+export const Home: FC<IHome> = (props) => {
```
Comment (anonymoussprocket) [inline V2/tezex/src/pages/Home/index.tsx:18]: consider adding some sort of standard formatting like prettier

File: V2/tezex/src/components/nav/NavHome.tsx [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+import { useNavigate } from "react-router-dom";
+import Tabs from "@mui/material/Tabs";
+import Tab from "@mui/material/Tab";
+import {useSession} from "../../hooks/session";
+
+import {
+	TransactingComponent,
+} from "../../types/general";
+const tabClasses = {
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/nav/NavHome.tsx]: separate styles

File: V2/tezex/src/components/removeLiquidity/style.js [hunk 9]
```diff
@@ -0,0 +1,337 @@
+
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/removeLiquidity/style.js]: should be a .ts rather than .js like the rest

File: V2/tezex/src/components/ui/elements/Buttons.tsx [hunk 10]
```diff
@@ -0,0 +1,90 @@
+import { FC, useEffect, useState } from "react";
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/ui/elements/Buttons.tsx]: rename the file to something more meaningful

File: V2/tezex/src/components/ui/elements/Labels.tsx [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import { FC, useEffect, useState } from "react";
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/ui/elements/Labels.tsx]: rename the file to something more meaningful

File: V2/tezex/src/components/ui/elements/inputs/TokenAmount/index.tsx [hunk 12]
```diff
@@ -0,0 +1,409 @@
+import {
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/ui/elements/inputs/TokenAmount/index.tsx]: filename, format, separate style

File: V2/tezex/src/components/wallet/Wallet.tsx [hunk 13]
```diff
@@ -1,34 +1,291 @@
...
+			props.transaction &&
+			props.transaction.sendAmount[0].decimal.eq(0)
+		) {
+			setSpinner(false);
+			return "Enter Amount";
+		} else if (props.transaction) {
+			const transactionStatus: TransactionStatus =
+				props.transaction.transactionStatus;
+			switch (transactionStatus) {
+				case TransactionStatus.INSUFFICIENT_BALANCE:
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/wallet/Wallet.tsx]: does this need three states?

File: V2/tezex/src/constants/index.tsx [hunk 14]
```diff
@@ -0,0 +1,57 @@
+import { TokenKind, Asset, Transaction } from "../types/general";
+import tzbtcLogo from "../assets/tzbtcLogo.svg";
+import tzlogo from "../assets/tzlogo.svg";
+import sirsLogo from "../assets/sirsLogo.svg";
+export const assets: Asset[] = [
```
Comment (anonymoussprocket) [inline V2/tezex/src/constants/index.tsx]: this should be in a config file

File: V2/tezex/src/constants/index.tsx [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+];
+
+export const XTZ: Asset = assets[0];
+export const TzBTC = assets[1];
+export const SIRS = assets[2];
+
+export const getDex = (
+	transaction: Transaction
+): string => {
+	const sirius = "KT1TxqZ8QtKvLu3V3JH7Gx58n7Co8pgtpQU5";
```
Comment (anonymoussprocket) [inline V2/tezex/src/constants/index.tsx]: this should be in a config file

File: V2/tezex/src/constants/index.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+	return sirius;
+	//Todo
```
Comment (anonymoussprocket) [inline V2/tezex/src/constants/index.tsx]: is this still relevant?

File: V2/tezex/src/contexts/network.tsx [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+export type NetworkMap = {
+	[network: string]: NetworkInfo;
+};
+export interface INetwork {
+	selectedNetwork: NetworkType;
+	networks: NetworkMap;
+}
+
+export const mainnet = {
+	network: NetworkType.MAINNET,
```
Comment (anonymoussprocket) [inline V2/tezex/src/contexts/network.tsx]: this should be in a config file

File: V2/tezex/src/contexts/network.tsx [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+	tezosServer: "https://mainnet.api.tez.ie",
+	tezosServerForTaquito: "",
+	apiKey: "ab682065-864a-4f11-bc77-0ef4e9493fa1",
```
Comment (anonymoussprocket) [inline V2/tezex/src/contexts/network.tsx]: this should be in a secrets file

Comment (SiddharthV1) [reply V2/tezex/src/contexts/network.tsx]: This should never have been committed(apiKey), its not even being used , it was a mistake , should the server also be in a secrets or just a config file?

File: V2/tezex/src/App.tsx (hunk 1 above)
Comment (SiddharthV1) [reply V2/tezex/src/App.tsx]: acknowledged

File: V2/tezex/src/components/addLiquidity/index.tsx_ (hunk 2 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/addLiquidity/index.tsx_]: no, I'll remove it

File: V2/tezex/src/components/nav/NavHome.tsx (hunk 8 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/nav/NavHome.tsx]: acknowledged

File: V2/tezex/src/components/removeLiquidity/style.js (hunk 9 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/removeLiquidity/style.js]: acknowledged

File: V2/tezex/src/components/swap/style.tsx (hunk 3 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/swap/style.tsx]: yes styling is a bit of a mess, some things changed in mui for example useStyles does not work anymore it got depreciated ,I have not spent time at figuring a work around but i will do that this week and hopefully then I can consolidate the styles.

File: V2/tezex/src/components/ui/elements/Buttons.tsx (hunk 10 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/ui/elements/Buttons.tsx]: acknowledged

File: V2/tezex/src/components/ui/elements/Labels.tsx (hunk 11 above)
Comment (SiddharthV1) [reply V2/tezex/src/components/ui/elements/Labels.tsx]: acknowledged

File: V2/tezex/src/constants/index.tsx (hunk 14 above)
Comment (SiddharthV1) [reply V2/tezex/src/constants/index.tsx]: acknowledged

File: V2/tezex/src/constants/index.tsx (hunk 15 above)
Comment (SiddharthV1) [reply V2/tezex/src/constants/index.tsx]: acknowledged

File: V2/tezex/src/contexts/network.tsx (hunk 17 above)
Comment (SiddharthV1) [reply V2/tezex/src/contexts/network.tsx]: acknowledged

File: V2/tezex/src/contexts/network.tsx (hunk 4 above)
Comment (SiddharthV1) [reply V2/tezex/src/contexts/network.tsx]: acknowledged

File: V2/tezex/src/pages/Home/index.tsx (hunk 7 above)
Comment (SiddharthV1) [reply V2/tezex/src/pages/Home/index.tsx:18]: I am using prettier but it formats to about 75 characters line length , what should I keep the line length to ? plus I can add it as a dev dependency  and auto format on build

File: V2/tezex/src/pages/removeLiquidity/index.tsx (hunk 6 above)
Comment (SiddharthV1) [reply V2/tezex/src/pages/removeLiquidity/index.tsx]: no

File: V2/tezex/src/components/addLiquidity/index.tsx [hunk 19]
```diff
@@ -0,0 +1,620 @@
...
+	useEffect(() => {
+		const updateTransactionBalance = async () => {
+			await updateBalance();
+		};
+
+		const interval = setInterval(
+			() => {
+				updateTransactionBalance();
+			},
+			loadingBalances ? 2000 : 5000
```
Comment (anonymoussprocket) [inline V2/tezex/src/components/addLiquidity/index.tsx]: when is this timeout used? does it impact the entire UI refresh?

File: V2/tezex/src/contexts/wallet.tsx [hunk 20]
```diff
@@ -1,14 +1,902 @@
...
+
+	const fetchTransaction = (id: string): Transaction | undefined => {
+		return transactions.find((t: Transaction) => t.id === id);
+	};
 
+	const getBalanceOfAssets = useCallback(
+		async (assets: AssetOrAssetPair): Promise<Amount | null> => {
+			if (toolkit && address) {
+				switch (assets.length) {
+					case 1:
```
Comment (anonymoussprocket) [inline V2/tezex/src/contexts/wallet.tsx]: enums would make this clearer

File: V2/tezex/src/functions/beacon.ts [hunk 21]
```diff
@@ -1,15 +1,131 @@
...
+		} else {
+			walletInfo.setAddress(activeAccount.address);
+		}
+
+		// signer provider ??
+	} catch (error) {
+		err = true;
+	} finally {
+		if (err) {
+			walletInfo.setClient(null);
```
Comment (anonymoussprocket) [inline V2/tezex/src/functions/beacon.ts]: logging to console may help users

File: V2/tezex/src/functions/transactions.ts [hunk 22]
```diff
@@ -0,0 +1,156 @@
...
+					toolkit
+				);
+		}
+	}
+};
+
+export const decimals = {
+	XTZ: 6,
+TzBTC: 8,
+	Sirius: 0,
```
Comment (anonymoussprocket) [inline V2/tezex/src/functions/transactions.ts]: is this a duplicate?

File: V2/tezex/src/functions/util.ts [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+	tokenMantissaToDecimal,
+} from "./scaling";
+
+
+export const shorten = (first: number, last: number, str: string) => {
+  return str.substring(0, first) + "..." + str.substring(str.length - last);
+};
+
+export function tezToMutez(amount: BigNumber) {
+	return amount.multipliedBy(1000000);
```
Comment (anonymoussprocket) [inline V2/tezex/src/functions/util.ts]: this function appears elsewhere

File: V2/tezex/src/hooks/wallet.ts [hunk 24]
```diff
@@ -1,8 +1,291 @@
...
+	TransactionStatus,
+	TransactingComponent,
+	AssetOrAssetPair,
+} from "../types/general";
+
+export interface WalletOps {
+	initialize: (
+		sendAsset: AssetOrAssetPair,
+		recieveAsset: AssetOrAssetPair,
+		sendAmount?: BigNumber, //TODO
```
Comment (anonymoussprocket) [inline V2/tezex/src/hooks/wallet.ts]: are these TODOs still relevant?

**Review by anonymoussprocket — APPROVED** [review]

**Review by vishakh — APPROVED** [review]
